app/src/main/java/eu/kanade/tachiyomi/ui/setting/SettingsAdvancedController.kt

```diff
@@ -110,12 +123,57 @@ class SettingsAdvancedController : SettingsController() {
         }
     }
 
-    private fun clearChapterCache() {
+    private fun getChaperCacheSize(): String {
+        val dirCache = GlideApp.getPhotoCacheDir(activity!!)
+        val realSize1 = DiskUtil.getDirectorySize(dirCache!!)
+        val realSize2 = DiskUtil.getDirectorySize(coverCache.cacheDir)
+        return Formatter.formatFileSize(activity!!, realSize1 + realSize2)
+    }
+
+    private fun clearImageCache() {
         if (activity == null) return
-        val files = chapterCache.cacheDir.listFiles() ?: return
 
+        // Clear the glide disk cache for our chapters
+        // Note: small hack since we require the glide clear to be in a background thread
+        // https://stackoverflow.com/a/46292711
+        val thread = Thread(Runnable {
```

Comment:
dont use threading.  Coroutine only

Reply:
Based it off the logic in `cleanupDownloads()` function.